### PR TITLE
Update publish GHA workflow to run smoke tests and promote images

### PR DESCRIPTION
**Pull request by rodesai**
Update publish GHA workflow to run smoke tests and promote images. I tested this
in my own fork of this repo to avoid messing with the release tags here.

### CODE REVIEW

File: .github/workflows/github-publish.yaml [hunk 1]
```diff
@@ -61,14 +61,77 @@ jobs:
...
+
+      - name: Set Operator Version
+        run: |
+          echo "OPERATOR_VERSION=$(./gradlew operator:cV | grep Project.version | sed 's/Project version: //')" >> $GITHUB_ENV
+
+      - name: Set Kafka Client Version
+        run: |
+          echo "KAFKA_CLIENTS_VERSION=$(./gradlew kafka-client:cV  | grep Project.version | sed 's/Project version: //')" >> $GITHUB_ENV
+
+      - name: Run Smoke Test (Operator)
```
Comment (agavra) [inline .github/workflows/github-publish.yaml:111]: if either of these fail, we probably want to "unpublish" the images, right? or is there a way for us not to publish them to the public location at all until after we've tested them?

**Review by agavra — APPROVED** [review]
lgtm, one comment inline

**Comment by agavra** [comment]
hmm looks like `dev.responsive.kafka.store.ResponsiveGlobalStoreIntegrationTest` is flaky. i'll take a look at that

File: .github/workflows/github-publish.yaml (hunk 1 above)
Comment (rodesai) [reply .github/workflows/github-publish.yaml:111]: yeah I was thinking about this and landed on either:
- we can just cut a new release
- we can setup a release patching process.
I think either approach is fine for where we're at now.

I swear I did consider running these before publishing but now I can't remember why I didn't implement it that way. Something didn't work when I tried. Let me think about it. I agree that would be preferable.

Comment (rodesai) [reply .github/workflows/github-publish.yaml:111]: I'll try to make it work. I'm just going to merge this first. And then do a follow-up with that change.

**Pull request merged**